packages/sdk/react-native/src/hooks/useBoolVariation.ts

```diff
@@ -0,0 +1,13 @@
+import useLDClient from './useLDClient';
+
+export const useBoolVariation = (flagKey: string, defaultValue: boolean) => {
+  const ldClient = useLDClient();
+  return ldClient?.boolVariation(flagKey, defaultValue) ?? defaultValue;
+};
+
+export const useBoolVariationDetail = (flagKey: string, defaultValue: boolean) => {
+  const ldClient = useLDClient();
+  return ldClient?.boolVariationDetail(flagKey, defaultValue) ?? defaultValue;
```

Comment:
Because detailed variations return objects we need to ensure we keep stable object references. Which may happen for flags that exist, because we might be returning a value we have stored.

I am concerned that the variation for a default value will not be stable. Which could lead to the hook triggering more re-renders than desired.

Reply:
Ok this is definitely a bug. What do you think of a fix like this:
```suggestion
  const def = {
    value: defaultValue,
    variationIndex: null,
    reason: null,
  };
  return ldClient?.boolVariationDetail(flagKey, defaultValue) ?? def;
```